src/components/form-events/index.tsx

```diff
@@ -0,0 +1,120 @@
+'use client';
+import { Button, DatePicker, Form, Input, Radio } from 'antd';
+import { useForm, Controller, FieldErrors } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import moment from 'moment';
+import { createValidObject } from './validation';
+import { events } from './mock/events';
+import { useFilterCalendarDate } from './hooks/filterCalendarDate';
+
+export const FormEvents = () => {
+    const fields: string[] = [
+        'UserName',
+        'UserNameTelegram',
+        'CalendarDate',
+        'TimeSlot',
+    ];
+
+    const yupSchema = yup.object().shape(createValidObject(fields));
+
+    const form = useForm({
+        defaultValues: {},
+        mode: 'all',
+        resolver: yupResolver(yupSchema),
+    });
+    const {
+        control,
+        handleSubmit,
+        formState: { errors },
+        watch,
+    } = form;
+
+    const onSubmit = (data: unknown) => {
+        console.log('data', data);
+    };
+
+    const createValidStatus = (errors: FieldErrors, fieldName: string) => {
+        return errors[fieldName] ? 'error' : 'success';
+    };
+
+    const createMessageValid = (errors: FieldErrors, fieldName: string) => {
+        const error = errors[fieldName];
+        return error && typeof error.message === 'string' ? error.message : '';
+    };
+
+    const [filteredEvents] = useFilterCalendarDate(watch(fields[2]), events);
+
+    return (
+        <Form layout="vertical" onFinish={handleSubmit(onSubmit)}>
```

Comment:
Take a look here 
https://javascript.plainenglish.io/combining-react-hook-form-and-ant-designs-input-component-for-efficient-form-handling-7606a38ff7c1

Also here https://danilorivera95.medium.com/make-your-first-form-with-react-hook-form-and-ant-design-acbea474f8f1